### ISSUE TITLE
It does not catch 'nodemon' reload.

### ISSUE THREAD

**Issue by Zirafnik** [sindresorhus/exit-hook#31]
The hook does not catch 'nodemon' reloads. I am not sure if this is by design or not.

**Comment by jpmnteiro** [comment]
@Zirafnik `nodemon` sends a `SIGUSR2` by default (see https://github.com/remy/nodemon/issues/1098).

Looking at the source code for this package, it doesn't seem to register a handler for that signal, so I suspect it's relying on the `exit` (or `beforeExit`) hooks.

**Comment by jakobo** [comment]
> Looking at the source code for this package, it doesn't seem to register a handler for that signal, so I suspect it's relying on the exit (or beforeExit) hooks.

That's correct. As Sindre said, `SIGUSR2` is a userland signal, which means we cannot act on it. The signal data doesn't include parent processes either.

Technically `SIGTERM` is [supposed to be polite termination](https://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html). We could change [its handler](https://github.com/sindresorhus/exit-hook/blob/a32478a214250c51f965447574712a854992531c/index.js#L76) to an asynchronous one without breaking existing implementations.

You could then change the signal with `nodemon --signal`, though your `nodemon` command will need a sufficient `--delay` flag in order to account for process cleanup.

Alternatively, `nodemon` could support `--signal message` which would send the signal identifier via IPC. This is how [PM2 works](https://pm2.keymetrics.io/docs/usage/signals-clean-restart/#windows-graceful-stop) in environments that don't support signaling. We then handle `process.on("message", ...)` and can shut down gracefully. Obviously it's easier to reinterpret `SIGTERM` than it is to change `nodemon`.
